### PR TITLE
MatGSOInterface support for Enumeration and bkz

### PR DESCRIPTION
**Pull request by WvanWoerden**
-Enumeration and bkz now accept a MatGSOInterface.
-ExactErrorBoundedEvaluator now uses a MatGSOInterface instead of a basis.
-Added a shortest_vector_ex function that accepts a MatGSOInterface
-Added test_svp_gram to test all of this.

### CODE REVIEW

File: tests/test_svp_gram.cpp [hunk 1]
```diff
@@ -0,0 +1,372 @@
+/* Copyright (C) 2015 Martin Albrecht
```
Comment (malb) [inline tests/test_svp_gram.cpp]: Change copyright

**Comment by malb** [comment]
Can you also add BKZ tests?

**Comment by WvanWoerden** [comment]
Basic BKZ tests are now done.

**Comment by WvanWoerden** [comment]
Ready for review when Travis is done.

File: fplll/enum/evaluator.h [hunk 2]
```diff
@@ -369,8 +369,8 @@ class ExactErrorBoundedEvaluator : public ErrorBoundedEvaluator
 
 private:
   FP_NR<mpfr_t> int_dist2Float(Z_NR<mpz_t> int_dist);
-
-  const ZZ_mat<mpz_t> &matrix;  // matrix of the lattice
+  MatGSOInterface<Z_NR<mpz_t>, FP_NR<mpfr_t>> &gso;
+  // const ZZ_mat<mpz_t> &matrix;  // matrix of the lattice
```
Comment (malb) [inline fplll/enum/evaluator.h]: maybe delete?

File: fplll/svpcvp.h [hunk 3]
```diff
@@ -42,6 +43,29 @@ int shortest_vector_pruning(ZZ_mat<mpz_t> &b, vector<Z_NR<mpz_t>> &sol_coord,
...
+
+int shortest_vector(MatGSOInterface<Z_NR<mpz_t>, FP_NR<mpfr_t>> &gso,
+                    vector<Z_NR<mpz_t>> &sol_coord, SVPMethod method = SVPM_PROVED,
+                    int flags = SVP_DEFAULT);
+
+int shortest_vectors(MatGSOInterface<Z_NR<mpz_t>, FP_NR<mpfr_t>> &gso,
+                     vector<vector<Z_NR<mpz_t>>> &sol_coord, vector<enumf> &sol_dist,
+                     const int max_sols, SVPMethod method, int flags);
+
+int shortest_vector_pruning(MatGSOInterface<Z_NR<mpz_t>, FP_NR<mpfr_t>> &gso,
```
Comment (malb) [inline fplll/svpcvp.h]: Do we need `_pruning`, doesn't the signature take care of this?

**Comment by malb** [comment]
Works with current fpylll master

**Pull request merged**